ts/components/wallet/TransactionsList.tsx

```diff
@@ -33,6 +33,7 @@ type Props = Readonly<{
   transactions: pot.Pot<ReadonlyArray<Transaction>, Error>;
   navigateToTransactionDetails: (transaction: Transaction) => void;
   ListEmptyComponent?: React.ReactNode;
+  unreadTransactions: ReadonlyArray<Transaction>;
```

Comment:
Instead of using the unreadTransactions array use the readTransaction object you have created in the redux store.